config/environments/staging.rb

```diff
@@ -0,0 +1,129 @@
+    .svg .eot .woff .ttf
+
+    admin-layout.css admin-layout.js
+    dashboard-layout.css dashboard-layout.js
+    basic-layout.css basic-layout.js
+    blank-layout.css basic-layout.js
+
+    admin.css admin.js
+    dashboard.css dashboard.js
+    home.css home.js
+    rosters.css rosters.js
+    requests.css requests.js
+    offers.css offers.js
+    outreach_patrols.css outreach_patrols.js
+    patrols.css patrols.js
+    proficiencies.css proficiencies.js
+    registrations.css registrations.js
+    selected_user.css selected_user.js
+    users.css users.js
+    reports.css
+  ]
+
+  # Generate digests for assets URLs.
+  config.assets.digest = true
+
+  # Version of your assets, change this if you want to expire all your assets.
+  config.assets.version = '1.1'
+
+  # Specifies the header that your server uses for sending files.
+  # config.action_dispatch.x_sendfile_header = "X-Sendfile" # for apache
+  # config.action_dispatch.x_sendfile_header = 'X-Accel-Redirect' # for nginx
+
+  # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
+  config.force_ssl = true
+
+  # Set to :debug to see everything in the log.
+  config.log_level = :debug
+
+  # Prepend all log lines with the following tags.
+  # config.log_tags = [ :subdomain, :uuid ]
+  config.log_tags = [:uuid]
+
+  # Use a different logger for distributed setups.
+  # config.logger = ActiveSupport::TaggedLogging.new(SyslogLogger.new)
+
+  # Use a different cache store in production
+  config.cache_store = :redis_store, ENV['REDIS_URL']
+
+  # Enable serving of images, stylesheets, and JavaScripts from an asset server.
+  # config.action_controller.asset_host = "https://www.swapsea.com.au"
+
+  # Precompile additional assets.
+  # application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
+  # config.assets.precompile += %w( search.js )
+
+  # Ignore bad email addresses and do not raise email delivery errors.
+  # Set this to true and configure the email server for immediate delivery to raise delivery errors.
+  # config.action_mailer.raise_delivery_errors = false
+
+  # Enable locale fallbacks for I18n (makes lookups for any locale fall back to
+  # the I18n.default_locale when a translation can not be found).
+  config.i18n.fallbacks = [I18n.default_locale]
+
+  # Send deprecation notices to registered listeners.
+  config.active_support.deprecation = :notify
+
+  # Disable automatic flushing of the log to improve performance.
+  # config.autoflush_log = false
+
+  # Use default logging formatter so that PID and timestamp are not suppressed.
+  config.log_formatter = ::Logger::Formatter.new
+
+  # ActionMailer Config
+  config.action_mailer.default_url_options = { host: ENV['ACTION_MAILER_HOST'] }
+  # Setup for production - deliveries, no errors raised
+  config.action_mailer.delivery_method = :smtp
+  config.action_mailer.perform_deliveries = (ENV['ACTION_MAILER_PERFORM_DELIVERIES'] == 'true')
```

Comment:
I think staging _should_ perform deliveries, but they should be re-routed to a dummy mailbox (as you suggested)

Reply:
I agree. I have looked at a couple of options to achieve the re-routing outcome but haven't found something that suits our workflow and budget.

Reply:
Ok, as long as we _never_ put the production database on staging.